app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -1191,4 +1218,11 @@ private void stopTimer() {
             }
         }
     }
+
+    public void onLocationChanged() {
+        boolean loaded = loadEntities();
+        if (!loaded) {
+            locationChangedWhileLoading = true;

```

Comment:
This should be persisted between device orientation changes, right? If the location is updated and you rotate, then loading completes, it currently won't retrigger entity loading.